src/sardana/macroserver/macros/standard.py

```diff
@@ -845,3 +845,27 @@ def run(self, nr):
                 self.__loop()
                 progress = ((i + 1) / float(nr)) * 100
                 yield progress
+
+
+class newfile(Macro):
+    """This macro sets a new ScanFile and ScanID values into the env.
+    The ScanID should be set to the value N of the last executed ScanID
+    in order to continue with a ScanID N+1 for the next scan."""
+
+    env = ('ScanFile', 'ScanID')
+
+    param_def = [
+        ['ScanID', Type.Integer, 1, 'Scan ID'],
```

Comment:
Do you know how spec deals the default value? If ones executes `newfile myfile.dat` does it reset the ScanID?

Reply:
I just tried it in SPEC and it sets the ScanID by default to 0 so the next scan starts with 1.
So this has to be implemented.